tac-operation-lifecycle/tac-operation-lifecycle-server/src/services/operations.rs

```diff
@@ -98,37 +114,40 @@ impl TacService for OperationsService {
     ) -> std::result::Result<tonic::Response<OperationsResponse>, tonic::Status> {
         let inner = request.into_inner();
 
-        match inner.q {
-            Some(q) => {
-                let operations = self.db.search_operations(&q).await.map_err(map_db_error)?;
+        let mut input_pagination = None;
+        if let Some(pagination_token) = inner.page_token {
+            input_pagination = Some(LogicPagination {
+                count: inner.page_items.unwrap_or(0) as usize,
+                earlier_timestamp: pagination_token,
+            });
+        }
```

Comment:
```suggestion
        let input_pagination = inner.page_token.map(|pagination_token|
        LogicPagination {
            count: inner.page_items.unwrap_or(0) as usize,
            earlier_timestamp: pagination_token,
        });
```

also maybe extract it somewhere to keep the endpoint definition as thin as possible

Reply:
Moved into the separate routine `extract_input_pagination` in https://github.com/blockscout/blockscout-rs/pull/1382/commits/f7500bd87a7970c5a048324ac35a8d51ce585a47